backend/eventloop/repocred_reconciler_test.go

```diff
@@ -0,0 +1,118 @@
+			Expect(err).To(BeNil())
+
+			// Create fake client
+			k8sClient = fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(apiNamespace, argocdNamespace, kubesystemNamespace).
+				Build()
+
+			err = db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			ctx = context.Background()
+			log = logger.FromContext(ctx)
+			dbq, err = db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+
+			By("Create required DB entries.")
+
+			_, _, _, gitopsEngineInstance, _, err = db.CreateSampleData(dbq)
+			Expect(err).To(BeNil())
+
+			// Create DB entry for ClusterUser
+			clusterUserDb = &db.ClusterUser{
+				Clusteruser_id: "test-repocred-user-id",
+				User_name:      "test-repocred-user",
+			}
+			err = dbq.CreateClusterUser(ctx, clusterUserDb)
+			Expect(err).To(BeNil())
+
+			// Create DB entry for RepositoryCredentials
+			gitopsRepositoryCredentialsDb = db.RepositoryCredentials{
+				RepositoryCredentialsID: "test-repo-" + string(uuid.NewUUID()),
+				UserID:                  clusterUserDb.Clusteruser_id,
+				PrivateURL:              "https://test-private-url",
+				AuthUsername:            "test-auth-username",
+				AuthPassword:            "test-auth-password",
+				AuthSSHKey:              "test-auth-ssh-key",
+				SecretObj:               "test-secret-obj",
+				EngineClusterID:         gitopsEngineInstance.Gitopsengineinstance_id,
+			}
+			err = dbq.CreateRepositoryCredentials(ctx, &gitopsRepositoryCredentialsDb)
+			Expect(err).To(BeNil())
+
+			// Create DB entry for APICRToDatabaseMapping
+			apiCRToDatabaseMappingDb = db.APICRToDatabaseMapping{
+				APIResourceType:      db.APICRToDatabaseMapping_ResourceType_GitOpsDeploymentRepositoryCredential,
+				APIResourceUID:       "test-" + string(uuid.NewUUID()),
+				APIResourceName:      "test-" + string(uuid.NewUUID()),
+				APIResourceNamespace: "test-" + string(uuid.NewUUID()),
+				NamespaceUID:         "test-" + string(uuid.NewUUID()),
+				DBRelationType:       db.APICRToDatabaseMapping_DBRelationType_RepositoryCredential,
+				DBRelationKey:        gitopsRepositoryCredentialsDb.RepositoryCredentialsID,
+			}
+
+			err = dbq.CreateAPICRToDatabaseMapping(ctx, &apiCRToDatabaseMappingDb)
+			Expect(err).To(BeNil())
+		})
+
+		It("It should reconcile status for RepositoryCredentials if the entry is present in RepositoryCredentials DB", func() {
+
+			defer dbq.CloseDatabase()
+
+			By("Call Reconcile function.")
+
+			reconcileRepositoryCredentials(ctx, dbq, k8sClient, log)
+
+			By("Verify that status is updated for GitopsDeploymentRepositoryCredentialCR.")
+			objectMeta := metav1.ObjectMeta{
+				Name:      apiCRToDatabaseMappingDb.APIResourceName,
+				Namespace: apiCRToDatabaseMappingDb.APIResourceNamespace,
+			}
+			repoCredCR := managedgitopsv1alpha1.GitOpsDeploymentRepositoryCredential{ObjectMeta: objectMeta}
+
+			if isOrphaned := isRowOrphaned(ctx, k8sClient, &apiCRToDatabaseMappingDb, &repoCredCR, log); isOrphaned {
```

Comment:
I believe this check should be removed. Based on code coverage data, it appears that these lines are causing the `Expect(...)` calls below not to run.